### ISSUE TITLE
Closing a pane does not reset the PowerShell text position

### ISSUE THREAD

**Issue by richardszalay** [microsoft/terminal#2432]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18362.175]
Windows Terminal version (if applicable): 0.3.2171.0
```

# Steps to reproduce

1. Create a PowerShell tab
2. Press enter until the text covers most of the window
3. Split the view horizontally
4. Close the bottom pane
5. Start typing

# Expected behavior

Text appears at previous cursor line

# Actual behavior

Text appears in blank space. Screenshots of process:

![pane-text-1](https://user-images.githubusercontent.com/229467/63062276-9da01300-bf3b-11e9-9545-9b90e42dfb23.png)

![pane-text-2](https://user-images.githubusercontent.com/229467/63062280-a0026d00-bf3b-11e9-9db6-e424cf0b4b5f.png)

![pane-text-3](https://user-images.githubusercontent.com/229467/63062283-a395f400-bf3b-11e9-8b1c-cd0c66e44bc1.png)

**Comment by zadjii-msft** [comment]
So this is _definitely_ a psreadline bug, though it does seem unique to Windows Terminal. Notably, I can't seem to repro a similar scenario in a conhost window:

1. Do a `wsl.exe powershell.exe` to get a conpty running powershell in a conhost window.
2. fill the buffer
3. Resize down so the prompt is in the middle of the buffer
4. Resize up again
5. type anything

If this was a purely psreadline or conpty bug, then this wouldn't repro in this scenario. We'll need to do something better with the Terminal specifically.

**Comment by DHowett-MSFT** [comment]
This is partially related to the fact that PowerShell saves and restores your cursor position across taking prompt input. That's also the cause of #2267.

**Comment by zadjii-msft** [comment]
Okay with #4741 merged, this is still happening. I'm pretty sure PsReadline is saving the cursor position at the start of the prompt, and restores the position when it wants to redraw the terminal. Though, the question I have is "_why isn't this happening if you just decrease, then increase the height of the terminal_"


<details>
<summary>This discussion from VTE seemed relevant, but I'm not sure </summary>


https://gitlab.gnome.org/GNOME/vte/blob/master/doc/rewrap.txt
```
Shell prompt
────────────

If you resize the terminal to be narrower than your shell prompt (plus the
command you're entering) while the shell is waiting for your command, you see
weird behavior there. This is not a bug in rewrapping: it's because the shell
redisplays its prompt (and command line) on every resize. There's not much VTE
could do here.

As a long term goal, maybe readline could have an option where it knows that
the terminal rewraps its contents so that it doesn't redisplay the prompt and
the command line, just expects the terminal to do this correctly. It's a bit
risky, since probably all terminals that support rewrapping do this a little
bit differently.

```

</details>

**Comment by zadjii-msft** [comment]
@lzybkr, @stevel-msft

Any ideas what might be going on here? I'm thinking that PSReadline is restoring the cursor _relative to the bottom of the viewport_, but I don't know where in PSReadline's code I'd go to check that out.

**Comment by lzybkr** [comment]
The cursor position is saved after the prompt [here](https://github.com/PowerShell/PSReadLine/blob/198a64390e713214df20ef580f7a546b5c3ac1e8/PSReadLine/ReadLine.cs#L696).

Console resizing has always a bit of problem, but mostly nobody noticed when the code was originally written because resizing used to be awkward. I considered moving away from .Net to INPUT_RECORD to get the resize events early in Win10 when resizing a console became much easier, but that would have made the code much less portable so I'm glad I didn't.

**Comment by lzybkr** [comment]
BTW, @daxian-dbw maintains PSReadLine now.

**Comment by zadjii-msft** [comment]
Okay I'm just pasting here for my future reference code that looks relevant
This issue: https://github.com/PowerShell/PSReadLine/issues/682
and the associated PR: https://github.com/PowerShell/PSReadLine/pull/694

Which pointed me to this function
https://github.com/PowerShell/PSReadLine/blob/198a64390e713214df20ef580f7a546b5c3ac1e8/PSReadLine/Render.cs#L985-L1000

I can't be sure that's actually it, but that's certainly something to look into.

It looks like I've got 2.0.0 on my powershell 5 instance, and 2.0.0-beta3 on my powershell 6.2.3 instance.

Though, reading through that, it looks like `_console.CursorTop` might already be wrong at that point in the call. Looks like that function is just trying to see how many rows the input line takes, then is setting the `_initialY` to be that many rows above the cursor. So I'd think the cursor was already wrong at this point.

I'll keep digging

**Comment by zadjii-msft** [comment]
A simpler repro from #5254

> # Steps to reproduce
> 
> 1. Open Terminal in full screen
> 2. Type `ls`
> 3. unmaximize the window
> 4. maximize the window
> 5. type `ls` again and see that ls is not positioned correctly
> 
> 
> ![displaybug](https://user-images.githubusercontent.com/18528548/78541017-45cd5980-77f5-11ea-95aa-d34eb2eef788.gif)

**Comment by zadjii-msft** [comment]
This may have been fixed in an updated PsReadline. Just need to confirm.

> 🎉 This issue was addressed in 3074, which has now been successfully released in [`v2.2.0-beta5`](https://github.com/PowerShell/PSReadLine/releases/tag/v2.2.0-beta5). 🎉

**Comment by zadjii-msft** [comment]
Just confirmed over here that this looks like it was fixed in PSReadline 🙌
